packages/react-portal/src/components/Portal/usePortal.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import {
+  makeMergeProps,
+  resolveShorthandProps,
+  useIsomorphicLayoutEffect,
+  useMergedRefs,
+} from '@fluentui/react-utilities';
+import { PortalProps, portalShorthandProps, PortalState } from './Portal.types';
+import { usePortalContext } from '../../portalContext';
+import { usePortalMountNode } from '../../usePortalMountNode';
+
+const mergeProps = makeMergeProps<PortalState>({ deepMerge: portalShorthandProps });
+
+/**
+ * Create the state required to render Portal.
+ *
+ * The returned state can be modified with hooks such as usePortalStyles,
+ * before being passed to renderPortal.
+ *
+ * @param props - props from this instance of Portal
+ * @param ref - reference to root HTMLElement of Portal
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ *
+ */
+export const usePortal = (props: PortalProps, ref: React.Ref<HTMLElement>, defaultProps?: PortalProps): PortalState => {
+  const state = mergeProps(
+    {
+      as: 'div',
+      ref: useMergedRefs(ref, React.useRef(null)),
```

Comment:
This looks really strange as in `renderPortal()` nothing renders `slots.root`. We should either cleanup `*Props` & `*State` interfaces or render `slots.root`.

Reply:
? `slots.root` is rendered into the portal